src/k8s/pkg/utils/k8s/configmap_test.go

```diff
@@ -0,0 +1,107 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/fake"
+	k8stesting "k8s.io/client-go/testing"
+)
+
+func TestWatchConfigMap(t *testing.T) {
+	ctx := context.Background()
+
+	tests := []struct {
+		name      string
+		configmap *corev1.ConfigMap
+	}{
+		{
+			name: "example configmap with values",
+			configmap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-config", Namespace: "kube-system"},
+				Data: map[string]string{
+					"non-existent-key1": "value1",
+					"non-existent-key2": "value2",
+					"non-existent-key3": "value3",
+				},
+			},
+		},
+		{
+			name: "configmap with empty data",
+			configmap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-config", Namespace: "kube-system"},
+				Data:       map[string]string{},
+			},
+		},
+		{
+			name:      "pass nil object",
+			configmap: nil,
+		},
+	}
+
+	clientset := fake.NewSimpleClientset()
+	watcher := watch.NewFake()
+	clientset.PrependWatchReactor("configmaps", k8stesting.DefaultWatchReactor(watcher, nil))
+
+	client := &Client{Interface: clientset}
+
+	var receivedMaps []*corev1.ConfigMap
+
+	go client.WatchConfigMap(ctx, "kube-system", "test-config", func(configMap *corev1.ConfigMap) error {
+		receivedMaps = append(receivedMaps, configMap)
+		if configMap == nil {
+			return fmt.Errorf("unexpected nil map test case error")
+		}
+		return nil
+	})
+
+	defer watcher.Stop()
+
+	for i, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			watcher.Add(tc.configmap)
+			time.Sleep(100 * time.Millisecond)
```

Comment:
Is this to make sure that the watch function has executed? If so, don't do a sleep, but rather use a channel.

```
doneCh := make(chan struct{})

// in the goroutine
go watch...(...) { 
   doneCh <-struct{}{}
}()

// then outside the loop, replace the time.Sleep with

select {
case <-doneCh:
case <-time.After(time.Second):
  t.Fatal("Timed out waiting for watch to complete")
}
```